### PR TITLE
Python3 compatibility

### PR DESCRIPTION
**Pull request by jloehel**
It runs in python2 and python3.

```
==================================================================================================================== test session starts =====================================================================================================================
platform linux -- Python 3.7.5, pytest-5.3.1, py-1.8.0, pluggy-0.13.1
rootdir: /home/jloehel/Projects/w3af-api-client, inifile: setup.cfg
plugins: cov-2.8.1
collected 7 items                                                                                                                                                                                                                                            

tests/test_exceptions.py .                                                                                                                                                                                                                             [ 14%]
tests/test_fuzzable_requests.py .                                                                                                                                                                                                                      [ 28%]
tests/test_log.py .                                                                                                                                                                                                                                    [ 42%]
tests/test_query_scans.py .                                                                                                                                                                                                                            [ 57%]
tests/test_scan.py .                                                                                                                                                                                                                                   [ 71%]
tests/test_stop_clear.py .                                                                                                                                                                                                                             [ 85%]
tests/test_urls.py .                                                                                                                                                                                                                                   [100%]

----------- coverage: platform linux, python 3.7.5-final-0 -----------
Name                                      Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------
w3af_api_client/__init__.py                   6      0   100%
w3af_api_client/connection.py                97     17    82%   15, 51-53, 68-69, 123, 127-131, 147-153, 164-165, 170
w3af_api_client/finding.py                    9      1    89%   18
w3af_api_client/log.py                       64     12    81%   37-40, 72-74, 80, 93-95, 101, 107
w3af_api_client/scan.py                      93     27    71%   31-33, 46-48, 54-55, 82-92, 107, 112, 121-123, 129, 138, 143, 152, 157, 162
w3af_api_client/scanner_exception.py          6      1    83%   14
w3af_api_client/traffic.py                   20      2    90%   29, 32
w3af_api_client/utils/__init__.py             1      0   100%
w3af_api_client/utils/constants.py            1      0   100%
w3af_api_client/utils/details.py             27      7    74%   15, 18, 27-31, 50-51
w3af_api_client/utils/exceptions.py          12      0   100%
w3af_api_client/utils/stdout_wrapper.py      10      3    70%   14-17, 20
-----------------------------------------------------------------------
TOTAL                                       346     70    80%


===================================================================================================================== 7 passed in 0.36s ======================================================================================================================
```

Signed-off-by: Jürgen Löhel <juergen@loehel.de>

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,6 @@
 # These requirements are only needed for dev/testing, the user requirements
 # are in setup.py
 #
-mock==1.0.1
-httpretty==0.8.3
-pylint==1.1.0
-nose==1.3.0
+future
```
Comment (andresriancho) [inline requirements.txt]: I prefer requirements with `==`

If in the future any of these libraries push a change which is not back-compatible our project might be affected.

File: devel-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-r requirements.txt
+
+responses
+pytest
+pytest-cov
+mock
+
+pylint
+flake8
+pep8
```
Comment (andresriancho) [inline devel-requirements.txt]: I prefer requirements with `==`

If in the future any of these libraries push a change which is not back-compatible our project might be affected.

File: tests/test_query_scans.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import json
-import httpretty
```
Comment (andresriancho) [inline tests/test_query_scans.py:2]: Why change the file location of all tests from `w3af_api_client/tests/` to `tests`?

**Comment by andresriancho** [comment]
This is a **very nice** PR, thanks for all the effort you put into it.

Just one comment about the whole PR: Group change groups in different PRs. In this case you sent a change from nose to pytest, httpretty to responses and py3k support. In this case I'll most likely accept all of those changes because I don't have a strong preference for nose / httpretty (at least not for this repo). But if that would be the case then I might reject the whole PR and that would kill the py3k support.

Please review the comments I made to the code and we'll merge this later.

File: requirements.txt (hunk 1 above)
Comment (jloehel) [reply requirements.txt]: Okay, no problem. I will freeze it.

File: tests/test_query_scans.py (hunk 3 above)
Comment (jloehel) [reply tests/test_query_scans.py:2]: I prefer to keep tests separate from actual application code. This has the advantage that your tests can run against an installed version.

**Comment by jloehel** [comment]
> This is a **very nice** PR, thanks for all the effort you put into it.
> 
> Just one comment about the whole PR: Group change groups in different PRs. In this case you sent a change from nose to pytest, httpretty to responses and py3k support. In this case I'll most likely accept all of those changes because I don't have a strong preference for nose / httpretty (at least not for this repo). But if that would be the case then I might reject the whole PR and that would kill the py3k support.
> 
> Please review the comments I made to the code and we'll merge this later.

Thanks, I needed the changes for the new faraday version and I'd like to see the changes also upstream. About httpretty, it is not python3 friendly and caused dependency issues. The next time I will explain this in the comment. About nose, I have not really a strong opinion on that. I can change it again, if you like.

**Comment by andresriancho** [comment]
Thanks for explaining the reason behind the migration to `responses` from `httpretty`

We can keep `pytest`

**Comment by andresriancho** [comment]
giphy (1)

**Comment by cript0nauta** [comment]
Nice to see Python 3 support in the Python security ecosystem! Could you bump a new version of the library so we use it for Faraday 3.10 (first version using Python 3)?

**Comment by andresriancho** [comment]
@cript0nauta done!

**Pull request merged**